File: spr2018_helpdesk-master/2125/NumericalAnalysis/NumericalAnalysis.java
package NumericalAnalysis;
public class NumericalAnalysis
{
    public double solveForZero(Function func, double low, double high)
    {
        double epsilon = 1;
        double mid = 0;
        while(Math.abs(epsilon) > 0.00000001)
        {
            mid = low + ((high - low) / 2);
            epsilon = func.f(mid);
            if(epsilon > 0){
                high = mid;
            } else if (epsilon < 0) {
                low = mid;
            }
        }
        return mid;
    }
}

File: spr2018_helpdesk-master/2125/NumericalAnalysis/c++/NumericalAnalysis.cc
#include "NumericalAnalysis.h"
#include <stdlib.h>
#include <cmath>
#include <iostream>
//using namespace std;

NumericalAnalysis::NumericalAnalysis( ){ }

NumericalAnalysis::~NumericalAnalysis( ){ }

double NumericalAnalysis::solveForZero(Function* func, double low, double high)
{
    double epsilon = 1;
    double mid = 0;
    while(fabs(epsilon) > 0.0000001)
    {
        mid = low + ((high - low) / 2.0);
        epsilon = func->f(mid);
        if(epsilon > 0){
            high = mid;
        } else if (epsilon < 0){
            low = mid;
        }
    }
    return mid;
}
